### PR TITLE
Issue #1215: Members with a hash in the name can't be shown if using APIML

### PR DESCRIPTION
**Pull request by katelynienaber**
## Proposed changes

Fixed by replacing `#` with `%23` in the HTTP request.

I tested it with APIML and regular old non-APIML, whatever you call that XD Seems to work for both. Please confirm!

## Release Notes

Milestone: Backlog
Changelog: Fixed error on APIML when showing a member with a hash in the name

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

**Comment by jellypuno** [comment]
@katelynienaber Can you please check the theia tests?

File: .github/actions/deploy-vscode-extension/index.js [hunk 1]
```diff
@@ -27,12 +27,9 @@ const checkVersion = (packageJson, version) => {
...
 // VSCE specific steps for publishing an extension
-// VSCE token contains two values: "<VSCode Marketplace token> <Open VSX token>"
-// ex: "s0m3-V3ry-L0ng-T0k3n an0th3r-L0ng-T0k3n"
-const publishSpecificProject = (versionName, token, packagePath) => {
-    console.log(`Publishing: ${versionName}`);
-    console.log(execSync(`vsce publish --yarn -p ${token.split(" ")[0]}`, { cwd: packagePath }).toString());
-    console.log(execSync(`ovsx publish ${versionName}.vsix -p ${token.split(" ")[1]}`, { cwd: "dist" }).toString());
+const publishSpecificProject = (version, token, packagePath) => {
+    console.log(`Publishing version ${version}`);
+    console.log(execSync(`vsce publish --yarn -p ${token}`, { cwd: packagePath }).toString());
```
Comment (jellypuno) [inline .github/actions/deploy-vscode-extension/index.js:32]: why is the token removed?

**Review by jellypuno — COMMENTED** [review]
same here. It should be a 1 file change but you have many files affected by formatting

**Comment by jellypuno** [comment]
@zFernand0 Will the changes in the deploy workflow trigger the publishing process?

**Comment by zFernand0** [comment]
> @zFernand0 Will the changes in the deploy workflow trigger the publishing process?

They should not. However, even if they do, nothing will get published as the `package.json#version` never changed 😋

**Comment by jellypuno** [comment]
Displaying the member with a hash worked for me but this introduced other bugs. 😆 Or at least it opened the door for me to test other functions. Rename and creating a member with # has the same error. I am now wondering what other special characters are not working. Maybe this should be really  handled in  APIML level rather than with us.

**Comment by jellypuno** [comment]
@katelynienaber If you agree, I would like to test this more. I think this problem has more connected issues and we need analyze it better. I would like to put this PR on-hold and investigate it further. I apologize for the delay in my review.

**Comment by lauren-li** [comment]
It seems this issue also occurs with opening data sets with a hash in the name. I was able to recreate the issue in Zowe CLI as well (when using a z/OSMF profile authenticated through API ML). Below is a snippet of the error I got back. I was able to execute the same command successfully using another z/OSMF profile that was connected directly to the same server:

Screen Shot 2021-04-20 at 3 14 16 PM

As it appears the issue is common to Zowe extensions that go through APIML, I agree with @jellypuno that it might make sense to have the APIML team take a look at this.

**Comment by MikeBauerCA** [comment]
This may be related -> https://github.com/zowe/api-layer/issues/394#issuecomment-537162316

**Comment by katelynienaber** [comment]
@jellypuno @lauren-li Yes that sounds like a good thing to do. Also it seems like quite often we have some issues with foreign characters, maybe we can look at special characters throughout the extension to see which ones work...

**Pull request closed without merging**